webapp/packages/plugin-resource-manager/src/SaveScriptDialog.tsx

```diff
@@ -0,0 +1,94 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2022 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+
+import { observable } from 'mobx';
+import { observer } from 'mobx-react-lite';
+import { useCallback } from 'react';
+import styled, { css } from 'reshadow';
+
+import { BASE_CONTAINERS_STYLES, Button, Container, InputField, SubmittingForm, useFocus, useObservableRef } from '@cloudbeaver/core-blocks';
+import { CommonDialogWrapper, DialogComponent } from '@cloudbeaver/core-dialogs';
+import { Translate, useTranslate } from '@cloudbeaver/core-localization';
+import { useStyles } from '@cloudbeaver/core-theming';
+
+const style = css`
+  fill {
+    flex: 1;
+  }
+`;
+
+interface Payload {
+  defaultScriptName?: string;
+}
+
+export const SaveScriptDialog: DialogComponent<Payload, string> = observer(function SaveScriptDialog({
+  payload,
+  resolveDialog,
+  rejectDialog,
+  className,
+}) {
+  const translate = useTranslate();
+  const [focusedRef] = useFocus<HTMLFormElement>({ focusFirstChild: true });
+
+  const state = useObservableRef(() => ({
+    value: payload.defaultScriptName,
+  }), {
+    value: observable.ref,
+  }, {
+    payload,
+  });
+
+  const handleChange = useCallback((value: string | undefined) => {
+    if (value !== undefined) {
+      state.value = value;
+    }
+  }, []);
```

Comment:
please use `state` or `onChange` in `InputField` currently this callback is extra